File: test-suite-kotlin/src/test/kotlin/io/micronaut/docs/metrics/exception/StockController.kt
package io.micronaut.docs.metrics.exception

import io.micronaut.context.annotation.Requires
import io.micronaut.http.annotation.Controller
import io.micronaut.http.annotation.Get

@Requires(property = "spec.name", value = "OutOfStockExceptionTest")
@Controller("/stock")
class StockController {

    @Get("/{name}")
    fun stock(name: String): Int {
        throw OutOfStockException()
    }
}

File: test-suite-python/src/test/python/micronaut/docs/metrics/exception/StockController.py
from micronaut.context.annotation import Requires
from micronaut.http.annotation import Controller, Get

from .OutOfStockException import OutOfStockException


@Requires(property="spec.name", value="OutOfStockExceptionTest")
@Controller("/stock")
class StockController:

    @Get("/{name}")
    def stock(self, name: str) -> int:
        raise OutOfStockException()
